addons/project_timesheet_holidays/models/hr_holidays.py

```diff
@@ -106,10 +106,23 @@ def _timesheet_prepare_line_values(self, index, work_hours_data, day_date, work_
             'company_id': self.holiday_status_id.timesheet_task_id.company_id.id or self.holiday_status_id.timesheet_project_id.company_id.id,
         }
 
+    def _check_missing_global_leave_timesheets(self):
+        for leave in self:
+            global_leaves = self.env['resource.calendar.leaves'].search([
+                ("resource_id", "=", False),
+                ("date_to", ">=", leave.date_from),
+                ("date_from", "<=", leave.date_to),
+                ("calendar_id.company_id.internal_project_id", "!=", False),
+                ("calendar_id.company_id.leave_timesheet_task_id", "!=", False),
+            ])
+            if global_leaves:
+                global_leaves._generate_public_time_off_timesheets(leave.employee_ids)
```

Comment:
We should do one *big* search instead of doing one search per iteration. 🙂 

To do that, we could find the min and max dates according to the leaves we get in `self` and do one search to gather all the global leaves in that interval.
Once it is done we could search for which employees we need to generate a global time off.

I hope everything is clear, if not let me know and I will explain it with some code to be more explicit. 🙂